app/scripts/controller-init/assets/token-rates-controller-init.ts

```diff
@@ -16,12 +16,15 @@ import { TokenRatesControllerMessenger } from '../messengers/assets';
 export const TokenRatesControllerInit: ControllerInitFunction<
   TokenRatesController,
   TokenRatesControllerMessenger
-> = ({ controllerMessenger, persistedState }) => {
+> = (request) => {
+  const { controllerMessenger, getController, persistedState } = request;
+  const preferencesController = () => getController('PreferencesController');
+
   const controller = new TokenRatesController({
     messenger: controllerMessenger,
     state: persistedState.TokenRatesController,
     tokenPricesService: new CodefiTokenPricesServiceV2(),
-    disabled: !persistedState.PreferencesController?.useCurrencyRateCheck,
+    disabled: !preferencesController().state?.useCurrencyRateCheck,
```

Comment:
Interesting. Could you expand on why we are reading for controller.state instead of persistedState?

Reply:
Yeah interesting.

The preferenceController is not using the init pattern yet, but has been passed as an existing controller.
https://github.com/MetaMask/metamask-extension/blob/4691f2f080220b266cace8c505e0b5a9ef2ad8ad/app/scripts/metamask-controller.js#L1838-L1878

I wonder if the `persistedState` does not contain the preferencesController state?